### DISCUSSION TITLE
Internationalization

### DISCUSSION THREAD

**Discussion by mainarthur** [forem/forem#20060] [Feature Requests]
It would be amazing to have internationalization feature in dev.to.

It will allow attracting new users from each corner of the world.

I see this huge feature in three smaller features:
1. **Have internationalization for UI**. Most common way to do this is to have a single-option select in **"Appearance"** settings
![mock](https://github.com/forem/forem/assets/67127683/0881040e-cfd0-458e-923b-db8454f2a4ea)
dev.to has a huge and diverse audience which could be involved in the UI translation process. I would personally like to add Ukrainian and Russian translations of UI. 
Also services like https://crowdin.com/, https://lokalise.com/ could be handful in this process
2. **Have internationalization for feed**. After having UI in many languages, it's a necessity to have feed internationalization options. The way I see it and how it's usually implemented is having two options:
 - **Have a feed language settings:**
![Feed languages settings](https://github.com/forem/forem/assets/67127683/0fc140b0-7baf-478e-bbf2-a5a79bed4928)
With these settings, users will be able to filter content by languages they know or want to read.
 - **Have a post language settings:**
![translate button](https://github.com/forem/forem/assets/67127683/4eb4ef2d-e79e-4870-a28d-1ac168626fd2)
Quick button for translating own posts to other languages
![language select](https://github.com/forem/forem/assets/67127683/af45008a-35f4-4c06-9c25-692a578eef53)
And language selector for post language

This is how I see this amazing feature which will help dev.to to gain new users and people from other countries use dev.to for blogging/reading.

**Comment by PhilipHow** [comment]
Hey @mainarthur thanks so much for your feature request and the effort gone into your mockups.

We're currently prioritizing internationalizing articles, over more broad support for changes to the UI. I hope we will return to prioritize the remaining elements of Ui internationalization but that is not our immediate priority.

Our plans for this are as follows:

* We recently shipped work to that [automatically detects the language of submitted articles](https://github.com/forem/forem/pull/19756)
* We're currently working on adapting our user model to support user language preferences
* We will be updating the user settings/preferences area to allow users to add which languages they understand
* We will be updating our feed logic to incorporate user's preferences when serving content to them.
* Finally, we'll be making some smaller changes to the UI to better serve non-English content.

We hope to complete this over the next month or so. Thanks so much for the mockups - we will definitely be referencing them for future design work in this area.

**Reply by mainarthur** [reply]
Thank you @PhilipHow for sharing plans and estimates! I hope my mocks will help and as I said I would be happy to help on creating localization files.
